fiona/ogrext.pyx

```diff
@@ -173,14 +175,26 @@ cdef class FeatureBuilder:
                 props[key] = ograpi.OGR_F_GetFieldAsInteger(feature, i)
             elif fieldtype is float:
                 props[key] = ograpi.OGR_F_GetFieldAsDouble(feature, i)
+
             elif fieldtype is text_type:
                 try:
                     val = ograpi.OGR_F_GetFieldAsString(feature, i)
-                    props[key] = val.decode(encoding)
+                    val = val.decode(encoding)
                 except UnicodeDecodeError:
                     log.warn(
                         "Failed to decode %s using %s codec", val, encoding)
-                    props[key] = val
+
+                # Does the text contain a JSON object? Let's check.
+                # Let's check as cheaply as we can.
+                if val.startswith('{'):
```

Comment:
@sgillies This feature is framed as being GeoJSON specific but it actually works with any driver.  Is this intentional?

On one hand I see this as another killer Fiona feature but also a potential source of confusion, especially because a mismatched field width will truncate the JSON object on write, making it impossible to decode.  Fiona speaks GeoJSON so it definitely makes sense from that perspective but AFAIK none of the other enabled drivers directly support complex fields.  Maybe some documentation is needed?